src/formatDate.js

```diff
@@ -8,7 +8,39 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
-}
+  const separatorFrom = fromFormat[fromFormat.length - 1];
+  const dateParts = date.split(separatorFrom);
+
+  const dateObj = {};
+
+  fromFormat.slice(0, -1).forEach((part, index) => {
+    dateObj[part] = dateParts[index];
+  });
+
+  const convertYear = (year, format) => {
```

Comment:
The `convertYear` function is embedded inside the `formatDate` function. It would be better to make it a separate function outside of `formatDate` for better readability and maintainability.